File: POO1/estiagem.cpp
// 1023
#include <bits/stdc++.h>
using namespace std;
using ll = long long;
using pl = pair<ll, ll>;

int main() {
  ios_base::sync_with_stdio(false);
  cin.tie(NULL);
  ll n, c{};
  while(cin >> n && n){
    map<ll, ll> q;
    ll soma{}, div{};
    for (ll i=0; i<n; i++){
      ll x, y; cin >> x >> y;
      soma+=y; div+=x;
      ll k = floor(y/x);
      q[k]+=x;
    }
    double media = (double)soma/div;
    media = floor(media*100)/100.0;

    cout << "Cidade# " << ++c << ":\n";
    for (auto &[k, v] : q){
      cout << v << '-' << k << ' ';
    }
    cout << '\n';
    cout << fixed << setprecision(2) << "Consumo medio: " << media << " m3.\n\n";
  }

  return 0;
}

File: POO1/estiagem.py
cidade=1
while True:
    n = int(input())
    if n==0:
        break
    soma, div =0,0
    pes_con = []
    for _ in range(n):
        x, y = map(int, input().split())
        soma+=y
        div+=x
        pes_con.append((y//x, x))
    media=soma/div

    pes_con = sorted(pes_con)
    print(f"Cidade# {cidade}:")
    cidade+=1
    for i in pes_con:
        x, y = i
        print(f"{y}-{x}", end=' ')
    print()
    print(f"Consumo medio: {media:.2f} m3.")
    print()
